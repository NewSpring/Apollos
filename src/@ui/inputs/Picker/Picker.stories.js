import React from 'react';

import { storiesOf } from '@storybook/react-native';
import FlexedView from '@ui/FlexedView';

import Picker, { Item } from './';

storiesOf('Inputs', module)
  .add('Picker', () => (
    <FlexedView>
      <Picker
        placeholder="Select a language..."
        label="Languages"
      >
        <Item label="Java" value="java" />
        <Item label="JavaScript" value="js" />
      </Picker>
    </FlexedView>
  ));
