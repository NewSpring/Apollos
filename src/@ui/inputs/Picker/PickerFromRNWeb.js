/* eslint-disable */
// TODO:
// This Picker was copy-pasta'd from react-native-web#master.
// We need to remove this and upgrade to the newest version of RNWeb when it is released.

/**
 * Copyright (c) 2017-present, Nicolas Gallagher.
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @providesModule Picker
 * @flow
 */

import applyNativeMethods from 'react-native-web/dist/modules/applyNativeMethods';
import { Component } from 'react';
import createElement from 'react-native-web/dist/modules/createElement';
import { StyleSheet } from 'react-native';
import { arrayOf, bool, func, number, oneOfType, string } from 'prop-types';

class PickerItem extends Component {
  static propTypes = {
    label: string.isRequired,
    testID: string,
    value: oneOfType([number, string]),
  };

  render() {
    const { label, testID, value } = this.props;
    return createElement('option', { label, testID, value });
  }
}

class Picker extends Component {
  static Item = PickerItem;

  render() {
    const {
      children,
      enabled,
      selectedValue,
      style,
      testID,
      /* eslint-disable */
      itemStyle,
      mode,
      prompt
      /* eslint-enable */
    } = this.props;

    return createElement('select', {
      children,
      disabled: enabled === false ? true : undefined,
      onChange: this._handleChange, // eslint-disable-line
      style: [styles.initial, style], // eslint-disable-line
      testID,
      value: selectedValue // eslint-disable-line
    });
  }

  _handleChange = (e: Object) => {
    const { onValueChange } = this.props; // eslint-disable-line
    const { selectedIndex, value } = e.target;
    if (onValueChange) {
      onValueChange(value, selectedIndex);
    }
  };
}

const styles = StyleSheet.create({
  initial: {
    fontFamily: 'inherit',
    fontSize: 'inherit',
    margin: 0,
  },
});

export default applyNativeMethods(Picker);
