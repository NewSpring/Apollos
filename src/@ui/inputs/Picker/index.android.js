import React from 'react';
import PropTypes from 'prop-types';
import { Animated, Picker as NativePicker } from 'react-native';
import { mapProps } from 'recompose';

import InputUnderline from '../InputUnderline';
import InputWrapper from '../InputWrapper';
import FloatingLabel from '../FloatingLabel';

const MappedNativePicker = mapProps(({
  placeholder: prompt,
  value: selectedValue,
  ...otherProps
}) => ({
  prompt,
  selectedValue,
  ...otherProps,
}))(NativePicker);

const Picker = ({ wrapperStyle, ...props }) => (
  <InputWrapper style={wrapperStyle}>
    <MappedNativePicker {...props} />
    <FloatingLabel animation={new Animated.Value(1)}>{props.label}</FloatingLabel>
    <InputUnderline />
  </InputWrapper>
);

Picker.propTypes = {
  label: PropTypes.string,
  wrapperStyle: PropTypes.any, // eslint-disable-line
};

export default Picker;
export const { Item } = NativePicker;
