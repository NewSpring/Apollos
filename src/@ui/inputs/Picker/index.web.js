import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure } from 'recompose';
import { Animated, Picker as NativePicker } from 'react-native';
import styled from '@ui/styled';

import InputUnderline from '../InputUnderline';
import InputWrapper from '../InputWrapper';
import FloatingLabel from '../FloatingLabel';
import withInputControlStyles from '../withInputControlStyles';

const enhance = pure;

const StyledNativePicker = compose(
  withInputControlStyles,
  styled({
    backgroundColor: 'transparent',
    borderWidth: 0,
    borderRadius: 0,
    outline: 'none',
  }),
)(NativePicker);

const Picker = enhance(({ wrapperStyle, value, ...props }) => (
  <InputWrapper style={wrapperStyle}>
    <StyledNativePicker {...props} selectedValue={value} prompt={props.placeholder} />
    <FloatingLabel animation={new Animated.Value(1)}>{props.label}</FloatingLabel>
    <InputUnderline />
  </InputWrapper>
));

Picker.propTypes = {
  label: PropTypes.string,
  placeholder: PropTypes.string,
  wrapperStyle: PropTypes.any, // eslint-disable-line
  value: PropTypes.any, // eslint-disable-line
};

export default Picker;
export const { Item } = NativePicker;
