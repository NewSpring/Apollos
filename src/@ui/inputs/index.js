export { default as Picker, Item as PickerItem } from './Picker';
export { default as Switch } from './Switch';
export { default as Text } from './Text';
export { default as DateInput } from './DateInput';
