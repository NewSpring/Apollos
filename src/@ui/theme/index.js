export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
