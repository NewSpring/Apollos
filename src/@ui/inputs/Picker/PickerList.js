import React from 'react';
import PropTypes from 'prop-types';
import { Modal, StyleSheet, View, Picker as NativePicker } from 'react-native';
import styled from '@ui/styled';
import { ButtonLink } from '@ui/Button';

const StyledPicker = styled(({ theme }) => ({
  borderTopWidth: StyleSheet.hairlineWidth,
  borderTopColor: theme.colors.background.inactive,
}), 'Inputs.Picker.List')(NativePicker);

const PickerKeyboardView = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
  borderTopColor: theme.colors.shadows.default,
  borderTopWidth: StyleSheet.hairlineWidth,
}), 'Inputs.Picker.List.Keyboard')(View);

const Toolbar = styled(({ theme }) => ({
  alignItems: 'flex-end',
  padding: theme.sizing.baseUnit / 2,
  backgroundColor: theme.colors.background.accent,
}), 'Inputs.Picker.List.Toolbar')(View);

const PickerList = ({
  focused,
  onRequestClose,
  value,
  ...pickerProps
}) => (
  <Modal
    visible={focused}
    onRequestClose={onRequestClose}
    animationType="slide"
    transparent
  >
    <PickerKeyboardView>
      <Toolbar>
        <ButtonLink onPress={onRequestClose}>Done</ButtonLink>
      </Toolbar>
      <StyledPicker selectedValue={value} {...pickerProps} />
    </PickerKeyboardView>
  </Modal>
);

PickerList.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  children: PropTypes.node,
  focused: PropTypes.bool,
  onRequestClose: PropTypes.func,
  mode: PropTypes.string,
};

export default PickerList;
