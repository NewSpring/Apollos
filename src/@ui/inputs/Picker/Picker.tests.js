import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@ui/theme';

import Picker, { Item } from './';

describe('The Picker Input component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Picker
          placeholder="Select a language..."
          label="Languages"
        >
          <Item label="Java" value="java" />
          <Item label="JavaScript" value="js" />
        </Picker>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
