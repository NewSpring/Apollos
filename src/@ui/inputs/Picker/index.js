import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Animated, Picker as NativePicker, TouchableOpacity } from 'react-native';
import styled from '@ui/styled';

import { H6 } from '@ui/typography';
import Icon from '@ui/Icon';

import FloatingLabel from '../FloatingLabel';
import InputUnderline from '../InputUnderline';
import InputWrapper from '../InputWrapper';
import withFocusAnimation from '../withFocusAnimation';

import InputAddon, { AddonRow } from '../InputAddon';
import withInputControlStyles from '../withInputControlStyles';

import PickerList from './PickerList';

const StyledH6 = withInputControlStyles(H6);
const Placeholder = styled(({ theme }) => ({
  color: theme.colors.input.placeholder,
}), 'Inputs.Picker.Placeholder')(H6);

class Picker extends PureComponent {
  static propTypes = {
    placeholder: PropTypes.string,
    displayValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    focusAnimation: PropTypes.any, // eslint-disable-line
    prefix: PropTypes.node,
    suffix: PropTypes.node,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    label: PropTypes.string,
    value: PropTypes.any, // eslint-disable-line
    wrapperStyle: PropTypes.any, // eslint-disable-line
    style: PropTypes.any, // eslint-disable-line
  };

  state = {
    focused: false,
  };

  toggle = () => {
    const focused = !this.state.focused;
    this.setState({ focused }, () => {
      if (focused) {
        this.props.onFocus();
      } else {
        this.props.onBlur();
      }
    });
  }

  render() {
    const {
      displayValue,
      focusAnimation, // from createInput
      placeholder,
      label,
      value,
      style,
      wrapperStyle,
      ...pickerProps
    } = this.props;
    const rotate = focusAnimation.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    const labelAnimation = value ? new Animated.Value(1) : focusAnimation;
    return (
      <InputWrapper style={wrapperStyle}>
        <TouchableOpacity onPress={this.toggle}>
          <AddonRow>
            <Animated.View style={{ opacity: labelAnimation, flex: 1 }}>
              <StyledH6 style={style}>
                {displayValue || (<Placeholder>{placeholder}</Placeholder>)}
              </StyledH6>
            </Animated.View>
            <InputAddon>
              <Animated.View style={{ transform: [{ rotate }] }}>
                <Icon name="arrow-down" size={18} />
              </Animated.View>
            </InputAddon>
          </AddonRow>
        </TouchableOpacity>

        <PickerList
          {...pickerProps}
          value={value}
          focused={this.state.focused}
          onRequestClose={this.toggle}
        />

        <FloatingLabel animation={labelAnimation}>{label}</FloatingLabel>
        <InputUnderline animation={focusAnimation} />
      </InputWrapper>
    );
  }
}

const EnhancedComponent = withFocusAnimation(Picker);
EnhancedComponent.propTypes = Picker.propTypes;

export default EnhancedComponent;
export const { Item } = NativePicker;
