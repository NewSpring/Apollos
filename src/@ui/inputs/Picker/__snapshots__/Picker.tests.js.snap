// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Picker Input component should render correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 20,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "opacity": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 15.75,
              "height": 40,
              "lineHeight": 30,
              "paddingBottom": 1.8,
              "paddingTop": 4.5,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            Select a language...
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <Svg
            height={18}
            viewBox="0 0 24 24"
            width={18}
          >
            <Path
              d="M12 18c-.16 0-.32-.06-.46-.2L3 9.66 3.9 8l8.1 7.72L20.1 8l.9 1.65-8.54 8.16c-.14.15-.3.2-.46.2z"
              fill="#303030"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderTopColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "rgba(221, 221, 221, 0.09999999999999998)",
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "color": "#6bac43",
              "textDecorationLine": "underline",
            }
          }
        >
          Done
        </Text>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "Java",
                "textColor": undefined,
                "value": "java",
              },
              Object {
                "label": "JavaScript",
                "textColor": undefined,
                "value": "js",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </Modal>
  <View
    collapsable={undefined}
    onLayout={[Function]}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "scale": 1,
          },
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#505050",
          "fontFamily": "Colfax-Bold",
          "fontSize": 15.75,
          "lineHeight": 17.75,
          "paddingBottom": 1.8,
          "paddingTop": 4.5,
          "paddingVertical": 5,
        }
      }
    >
      Languages
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#dddddd",
          "bottom": -0.5,
          "height": 0.5,
        }
      }
    />
    <View
      collapsable={undefined}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "transform": Array [
            Object {
              "scaleX": 0.0001,
            },
          ],
        }
      }
    >
      <View
        hasError={false}
        style={
          Object {
            "backgroundColor": "#6bac43",
            "bottom": 0,
            "height": 0.5,
          }
        }
      />
    </View>
  </View>
</View>
`;
